modules/core/src/main/scala/me/mnedokushev/zio/apache/parquet/core/codec/SchemaEncoderDeriver.scala

```diff
@@ -153,8 +153,20 @@ object SchemaEncoderDeriver {
       transform: Schema.Transform[A, B, ?],
       fields: => Chunk[Deriver.WrappedF[SchemaEncoder, ?]],
       summoned: => Option[SchemaEncoder[B]]
-    ): SchemaEncoder[B] = ???
+    ): SchemaEncoder[B] = summoned.getOrElse {
+      new SchemaEncoder[B] {
+        private def enc[A1](name0: String, schema0: Schema[A1], encoder: SchemaEncoder[?]) =
+          encoder.asInstanceOf[SchemaEncoder[A1]].encode(schema0, name0, isSchemaOptional(schema0))
+
+        override def encode(schema: Schema[B], name: String, optional: Boolean): Type = {
+          val fieldTypes = record.fields.zip(fields.map(_.unwrap)).map { case (field, encoder) =>
+            enc(field.name, field.schema, encoder)
+          }
 
+          Schemas.record(fieldTypes).optionality(optional).named(name)
+        }
+      }
+    }
```

Comment:
I will debug this one to figure out whether the transform should be involved here.

Reply:
@marcuswr I checked this. It is OK to do as you did because we don't use the `schema: Schema[B]` parameter during encoding to get the resulting value of type `Type`.